src/jit/lsra.cpp

```diff
@@ -5312,7 +5312,10 @@ LinearScan::spillInterval(Interval* interval, RefPosition* fromRefPosition, RefP
 
     if (!fromRefPosition->lastUse)
     {
-        if (!fromRefPosition->IsActualRef())
+        // If allocated a register, Lcl var def/use ref positions even if reg optional
+        // should be marked as spillAfter.
+        if (!fromRefPosition->RequiresRegister() && 
```

Comment:
I think the comment should be "If not allocated ...".
Also, you should add a note that, for a def of a lclVar, `resolveLocalRef` will simply store the result onto the stack, instead of spilling. However, for a use it will reload into a register, even if marked both `reload` and `spillAfter`. There should be a TODO comment and/or an issue that `resolveLocalRef` should be updated to use the lclVar directly from memory, instead of loading into a register, if it is marked `IsRegOptional`.